pkg/apis/core/validation/cloudprofile_test.go

```diff
@@ -996,6 +996,129 @@ var _ = Describe("CloudProfile Validation Tests ", func() {
+					Expect(errorList).To(BeEmpty())
+				})
+
+				It("should forbid unknown machineImage", func() {
+					cloudProfile.Spec.Bastion = &core.Bastion{
+						MachineImage: &core.BastionMachineImage{Name: "unknown"},
+					}
+
+					errorList := ValidateCloudProfile(cloudProfile)
+
+					Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal("spec.bastion.machineImage.name"),
+					}))))
+				})
+
+				It("should forbid preview image", func() {
+					cloudProfile.Spec.Bastion = &core.Bastion{
+						MachineImage: &core.BastionMachineImage{Name: "some-machineimage"},
+					}
+					cloudProfile.Spec.MachineImages[0].Versions[0].Classification = &previewClassification
+					cloudProfile.Spec.MachineImages[0].Versions[0].Architectures = []string{"amd64"}
+
+					errorList := ValidateCloudProfile(cloudProfile)
+
+					Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal("spec.bastion.machineImage.name"),
+					}))))
+				})
+
+				It("should forbid no arch images", func() {
+					cloudProfile.Spec.Bastion = &core.Bastion{
+						MachineImage: &core.BastionMachineImage{Name: "some-machineimage"},
+					}
+					cloudProfile.Spec.MachineImages[0].Versions[0].Classification = &supportedClassification
+					cloudProfile.Spec.MachineImages[0].Versions[0].Architectures = []string{}
+
+					errorList := ValidateCloudProfile(cloudProfile)
+
+					Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal("spec.bastion.machineImage.name"),
+					}))))
+				})
+
+				It("should allow images with supported classification and architecture specification", func() {
+					cloudProfile.Spec.Bastion = &core.Bastion{
+						MachineImage: &core.BastionMachineImage{Name: "some-machineimage"},
+					}
+					cloudProfile.Spec.MachineImages[0].Versions[0].Classification = &supportedClassification
+					cloudProfile.Spec.MachineImages[0].Versions[0].Architectures = []string{"amd64"}
+
+					errorList := ValidateCloudProfile(cloudProfile)
+					Expect(errorList).To(BeEmpty())
+				})
+
+				It("should allow matching arch of machineType and machineImage", func() {
+					cloudProfile.Spec.Bastion = &core.Bastion{
+						MachineType: &core.BastionMachineType{Name: machineType.Name},
+						MachineImage: &core.BastionMachineImage{
+							Name:    "some-machineimage",
+							Version: ptr.To("1.2.3"),
+						},
+					}
+					cloudProfile.Spec.MachineImages[0].Versions[0].Classification = &supportedClassification
+					cloudProfile.Spec.MachineImages[0].Versions[0].Architectures = []string{*machineType.Architecture}
+
+					errorList := ValidateCloudProfile(cloudProfile)
+					Expect(errorList).To(BeEmpty())
+				})
+
+				It("should forbid different arch of machineType and machineImage", func() {
```

Comment:
Could you add another positive test that allows this configuration, if `architecture` is not defined for the machine type, i.e. to cover [this case](https://github.com/gardener/gardener/blob/30815e241c9f2a81a2672169a2cd48a0c62c8306/pkg/apis/core/validation/cloudprofile.go#L330)?

Reply:
Added an additional test